src/todolist.js

```diff
@@ -0,0 +1,45 @@
+import './index.css';
+import {
+  getTodoList, addItem, removeItem, removeCompletedItems, editItem,
+} from './todolist.js';
+import displayToDoList from './dis.js';
+
+const inputItem = document.querySelector('.field-input');
+const addButton = document.querySelector('.add-btn');
+const clearAll = document.querySelector('#clear-btn');
+
+let editIndex = -1;
+addButton.addEventListener('click', () => {
+  if (editIndex === -1) {
+    addItem(inputItem.value);
+  } else {
+    editItem(editIndex.toExponential, inputItem.value);
+    editIndex = -1;
+    addButton.textContent = 'Add';
+  }
+  inputItem.value = '';
+  displayToDoList(getTodoList());
+});
+clearAll.addEventListener('click', () => {
+  removeCompletedItems();
+  displayToDoList(getTodoList());
+});
+
+// Event listener for removing an item
+document.addEventListener('click', (event) => {
+  if (event.target.classList.contains('remove-btn')) {
+    const itemIndex = parseInt(event.target.dataset.index, 10);
+    removeItem(itemIndex);
+    displayToDoList(getTodoList());
+  }
+});
+// Event listener for editing an item
+document.addEventListener('click', (event) => {
+  if (event.target.classList.contains('edit-btn')) {
+    const itemIndex = parseInt(event.target.dataset.index, 10);
+    inputItem.value = getTodoList()[itemIndex].description;
+    editIndex = itemIndex;
+    addButton.textContent = 'Save';
+  }
+});
```

Comment:
The code appears to be quite concise and focused on its purpose. However, here are some suggestions based on YAGNI and DRY principles:
- There are some duplicated lines of code. For instance, both the "Add" and "Save" buttons share some functionality, such as clearing the input field and displaying the updated list. You could extract that functionality into a separate function and call it from both places to avoid duplicating the code.
- The displayToDoList function could be called within each event listener to update the list immediately when a change occurs, rather than waiting until the end of the function to call it. This would make the code more responsive to user actions.